.github/workflows/assets.yml

```diff
@@ -86,13 +110,25 @@ jobs:
           for asset in assets/*; do
             mv "$asset" "assets/${{ env.ARCH }}.$(basename "$asset")"
           done
-      - name: Upload release files
-        id: upload-release-files
-        uses: softprops/action-gh-release@v2
+          md5sum "assets/${{env.ARCH}}.rootfs.img" > assets/"${{env.ARCH}}.rootfs.img.md5"
+      - name: Upload binary to GitHub Release
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          tag_name: ${{ inputs.tag_ref }}
-          make_latest: false
-          files: |
-            assets/${{ env.ARCH }}.*
+          RELEASE_ID: ${{ needs.create_release.outputs.release_id }}
+          UPLOAD_URL: ${{ needs.create_release.outputs.upload_url }}
+        run: |
+          for file in assets/*${{env.ARCH}}*; do
```

Comment:
Since you already converted all of the assets to `ARCH.*`, can you not just do `for file in assets/*`?

Reply:
Come to think of it, why even bother with the step "Rename files for release" and then this step? Just combine them:

```sh
for file in assets/*; do
    file_name=${{ env.ARCH }}.$(basename "$file")
    echo "Uploading $file as $file_name..."
    # do upload
done
```

Reply:
Good suggestion! We can combine them, but since `assets/${{ env.ARCH }}.rootfs.img.sha256` isn't part of the assets, I’ll just need to generate the sha256sum and make the necessary modifications.

Reply:
@deitch combined the two loops and added a condition for the sha256 checksum for rootfs.img

Reply:
Do the parallel runs in the matrix share their assets? If so the loop might pick up the assets from another run which isn't done yet.

Reply:
Nope, the parallel runs don't share the assets.

Reply:
ok